File: src/model/rice_cooker.rb

class RiceCooker

    attr_writer :is_operational
    attr_writer :is_cooking
    attr_writer :is_plugged

    def initialize(id, is_operational)
        @id = id
        @is_operational = is_operational
        @is_plugged = false
        @is_cooking = false
    end

    def id
        @id
    end

    def is_operational
        @is_operational
    end

    def is_plugged
        @is_plugged
    end

    def is_cooking
        @is_cooking
    end

    def to_s
        "Rice cooker : {
            id: #{id},
            is_operational: #{is_operational},
            is_cooking: #{is_cooking},
            is_plugged: #{is_plugged}
        }"
    end
end

File: src/main/model/RiceCooker.ts
export class RiceCooker {
  id: number
  isOperational: boolean
  isPlugged: boolean
  isCooking: boolean

  constructor (id: number, isOperational: boolean) {
    this.id = id
    this.isOperational = isOperational
    this.isCooking = false
    this.isPlugged = false
  }

  displayInfo (): void {
    console.log(`Rice cooker: {
            id: ${this.id},
            isOperational: ${this.isOperational.toString()},
            isPlugged: ${this.isPlugged.toString()},
            isCooking: ${this.isCooking.toString()}
        }`)
  }
}
